pkg/goDB/encoder/lz4cust/lz4cust_cgo.go

```diff
@@ -0,0 +1,96 @@
+//go:build cgo && !goprobe_nolz4cust
+// +build cgo,!goprobe_nolz4cust
+
+package lz4cust
+
+/*
+#cgo CFLAGS: -O3 -g
+#define LZ4_STATIC_LINKING_ONLY
+
+#include <stdlib.h>
+#include <stdio.h>
+#include "lz4hc.h"
+#include "lz4.h"
+
+int cCompressCust(int len, char *input, char *output, int level) {
+  return LZ4_compressHC2(input, output, len, level);
+}
+
+int cUncompress(char *output, int in_len, char *input, int out_len) {
+  return LZ4_decompress_safe(input, output, in_len, out_len);
+}
+*/
+import "C"
+
+import (
+	"errors"
+	"io"
+	"unsafe"
+)
+
+var (
+
+	// ErrDeprecated is used to signify the obsoletion of the lz4cust encoding
+	ErrDeprecated = error(nil)
```

Comment:
I didn't know you could cast a `nil`

Reply:
In this case you even have to, otherwise you can't use it (because `nil` isn't the same as `nil` :rofl:)...